pennylane_lightning/src/gates/cpu_kernels/avx_common/ApplyIsingXX.hpp

```diff
@@ -0,0 +1,175 @@
+            const auto v = PrecisionAVXConcept::load(arr + n);
+
+            const auto prod_cos = real_cos * v;
+            const auto prod_sin = imag_sin * Permutation::permute<perm>(v);
+
+            PrecisionAVXConcept::store(arr + n, prod_cos + prod_sin);
+        }
+    }
+    template <size_t min_rev_wire, class ParamT>
+    static void applyInternalExternal(std::complex<PrecisionT> *arr,
+                                      size_t num_qubits, size_t max_rev_wire,
+                                      bool inverse, ParamT angle) {
+        using namespace Permutation;
+
+        const size_t max_rev_wire_shift =
+            (static_cast<size_t>(1U) << max_rev_wire);
+        const size_t max_wire_parity = fillTrailingOnes(max_rev_wire);
+        const size_t max_wire_parity_inv = fillLeadingOnes(max_rev_wire + 1);
+
+        const double isin =
+            inverse ? std::sin(angle / 2) : -std::sin(angle / 2);
+        const auto cos_factor =
+            set1<PrecisionT, packed_size>(std::cos(angle / 2));
+        const auto isin_factor = imagFactor<PrecisionT, packed_size>(isin);
+
+        constexpr static auto perm = compilePermutation<PrecisionT>(
+            swapRealImag(flip(identity<packed_size>(), min_rev_wire)));
+
+        for (size_t k = 0; k < exp2(num_qubits - 1); k += packed_size / 2) {
+            const size_t i0 =
+                ((k << 1U) & max_wire_parity_inv) | (max_wire_parity & k);
+            const size_t i1 = i0 | max_rev_wire_shift;
+
+            const auto v0 = PrecisionAVXConcept::load(arr + i0);
+            const auto v1 = PrecisionAVXConcept::load(arr + i1);
+
+            const auto prod_cos0 = cos_factor * v0;
+            const auto prod_sin0 = isin_factor * Permutation::permute<perm>(v1);
+
+            const auto prod_cos1 = cos_factor * v1;
+            const auto prod_sin1 = isin_factor * Permutation::permute<perm>(v0);
+
+            PrecisionAVXConcept::store(arr + i0, prod_cos0 + prod_sin0);
+            PrecisionAVXConcept::store(arr + i1, prod_cos1 + prod_sin1);
+        }
+    }
+
+    template <class ParamT>
+    static void
+    applyExternalExternal(std::complex<PrecisionT> *arr,
+                          const size_t num_qubits, const size_t rev_wire0,
+                          const size_t rev_wire1, bool inverse, ParamT angle) {
+        using namespace Permutation;
+
+        const size_t rev_wire0_shift = static_cast<size_t>(1U) << rev_wire0;
+        const size_t rev_wire1_shift = static_cast<size_t>(1U) << rev_wire1;
+
+        const size_t rev_wire_min = std::min(rev_wire0, rev_wire1);
+        const size_t rev_wire_max = std::max(rev_wire0, rev_wire1);
+
+        const size_t parity_low = fillTrailingOnes(rev_wire_min);
+        const size_t parity_high = fillLeadingOnes(rev_wire_max + 1);
+        const size_t parity_middle =
+            fillLeadingOnes(rev_wire_min + 1) & fillTrailingOnes(rev_wire_max);
+
+        const double isin =
```

Comment:
As mentioned by @mlxd, could we use auto to account for both float & double parameters?

Reply:
Nice catch! Updated.